docs/website/contents/for-developers/Whitepaper.md

```diff
@@ -0,0 +1,244 @@
+
+In this section, we informally outline the requirements for Consensus layer of the node.
+
+Blocks can be theoretically forged every slot (e.g. every second on Cardano mainnet), so the Consensus layer must be
+extremely fast in producing the blocks once it knows it has to do so.
+
+As the Ouroboros family of protocols allows short-lived forks, for example due to slot battles in Praos, the Consensus layer has to be able to efficiently switch between these forks. This requires a mechanism for a fast rollback which is handled by the Storage Layer (see below).
+
+In terms of security, Consensus must not expose meaningful advantages for adversaries that could trigger a worst-case situation in which the amount of computation to be performed would block the node.
+This is generally achieved by trying to respect the following principle:
+
+- The cost of the worst case should be no greater than the cost of the average
+  case.
+
+We don't want to optimize for the average case because it exposes the node to DoS attacks.
+
+## Requirements imposed onto the Networking/Diffusion layer
+
+To maximize the probability of the block being included in the definitive chain,
+the Consensus layer has to strive to mint new blocks on top of the best block
+that exists in the network. Therefore it necessitates of a fast diffusion layer
+for blocks to arrive on time to the next block minters.
+
+Transmit chains as fast as possible so that blocks arrive to the next forger in time.
+
+## Requirements imposed onto the Ledger layer
+
+TODO
+
+# Single-era Consensus Layer
+
+In this section, we give a more detailed description of what _how_ the components of the reference Haskell implementation of the Consensus layer implement the requirements discussed in the previous section.
+
+## ChainSync client
+
+Each ChainSync client maintains an upstream peer's candidate chain.
+
+The protocol state is also independently maintained alongside each candidate chain by ChainSync.
+
+### Details
+
+It disconnects if the peer violates the mini protocol, if the peer sends an invalid header, or if switching to their chain would require rolling back more than kcp blocks of the local selection.
+
+*TODO*(georgy): what is kcp here?
+
+It's able to their validate headers past the intersection with the local selection because the parts of the ledger state necessary to validate a header were completely determined some positive number of slots ago on that header's chain; see _forecasting_ within the ledger rules and _snapshots_ within the ledger state.
+
+ChainSync blocks while the candidate chain is past the forecast range of the local selection.
+The forecast range must be great enough for the peer to rescue the local node from the worse-case scenario.
+The Praos paper bounds this to needing at most kcp+1 headers from the peer, which that paper also bounds to requiring at most scg slots of forecast range.
```

Comment:
`scg` is I think the stability window which on mainnet, by the Chain Growth Property of Praos is $3k/f=129600$ slots. (k = 2160, f = 0.05)

Reply:
I will use the term "stability window". This is probably already too much detail for this document though.